### ISSUE TITLE
Onboard shared process to --prof-start

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#117556]
We have the start profiler but only targets the main process, the renderer, and the extension host. Given that the shared process is becoming more critical we should also have (perf) insights about it

**Comment by jrieken** [comment]
fyi @bpasero

**Comment by bpasero** [comment]
Yeah makes sense. I also recently added an entry to the startup performance editor indicating the time it took to connect to the shared process. This will become more interesting when the shared process blocks the renderer from startup (if it needs, depends a bit on how we use it in the future).

If we need more precise timings, we probably need to add `perf.mark` into the shared process itself to drill deeper.

**Comment by jrieken** [comment]
This is actually tricky because the shared process is also a renderer...
